client/src/reducers/progress.js

```diff
@@ -1,28 +1,55 @@
 import * as Actions from "../actions/actionDefinitions";
```

Comment:
Good use of a separate reducer. Come to think of it, it probably makes sense to have roughly one per page.